assembly/Cargo.toml

```diff
@@ -17,30 +17,53 @@ edition.workspace = true
 bench = false
 doctest = false
 
+[[bench]]
+name = "deserialize_std_lib"
+harness = false
```

Comment:
I would probably put the benchmark into the `miden-vm` crate (i.e., [here](https://github.com/0xPolygonMiden/miden-vm/tree/main/miden/benches)). This is where we have the compilation and execution benchmarks, and I think deserialization benchmark could go there as well.

Reply:
If we put the benchmarks in `miden-vm` crate and use the approach from your comment [here](https://github.com/0xPolygonMiden/miden-vm/pull/1453#issuecomment-2293285367), then we would need to duplicate the build script from the `miden-stdlib` crate which compiles the `std.masl`. I have assumed we would prefer not to do that and instead put the benchmarks in the `miden-stdlib` crate.

If we don't want to put the benchmarks there and we don't want to run the build script in `miden-vm` crate, then the other option would be to take the approach @plafer explained [here](https://github.com/0xPolygonMiden/miden-vm/pull/1453#issuecomment-2293262299).

LMK what you guys prefer.

P.S. benchmark run example:
```
deserialize_std_lib/read_from_bytes
                        time:   [2.2909 ms 2.3113 ms 2.3379 ms]
                        change: [+2.9318% +6.2383% +11.557%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 11 outliers among 100 measurements (11.00%)
  1 (1.00%) low mild
  2 (2.00%) high mild
  8 (8.00%) high severe
```

Reply:
So, it takes about 2.3ms to deserialize `miden-stdlib`? Very cool! I wonder how it compares to the previous approach (my guess is that this is at least an order of magnitude faster).

I think one way to make benchmarking in the `miden-vm` relatively simple is to expose the serialized contents from the `StdLibrary` struct. For example, it could look like so:
```rust
impl StdLibrary {
    const SERIALIZED: &'static [u8] = include_bytes!(concat!(env!("OUT_DIR"), "/assets/std.masl"));
}

impl Default for StdLibrary {
    fn default() -> Self {
        Self(Library::read_from_bytes(Self::SERIALIZED).expect("failed to deserialize std masl!"))
    }
}
```
Then, in the benchmark, we'd be able to import the `StdLibrary` and then deserialize from `StdLibrary::SERIALIZED`.